components/mbedtls_cxx/include/mbedtls_wrap.hpp

```diff
@@ -1,28 +1,39 @@
 /*
- * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ * SPDX-FileCopyrightText: 2023-2024 Espressif Systems (Shanghai) CO LTD
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 #pragma once
 
 #include <utility>
 #include <memory>
-#include <span>
+#include <mbedtls/timing.h>
+#include <mbedtls/ssl_cookie.h>
 #include "mbedtls/ssl.h"
 #include "mbedtls/entropy.h"
 #include "mbedtls/ctr_drbg.h"
 #include "mbedtls/error.h"
 
-using  const_buf = std::span<const unsigned char>;
+using  const_buf = std::pair<const unsigned char *, std::size_t>;
+using  buf = std::pair<unsigned char *, std::size_t>;
+
+struct TlsConfig {
+    bool is_dtls;
+    uint32_t timeout;
+    const_buf client_id;
+};
 
 class Tls {
 public:
     enum class is_server : bool {};
     enum class do_verify : bool {};
+    enum class is_dtls : bool {};
```

Comment:
More description

Reply:
Added a common description to all of these basic config options